presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -596,6 +596,40 @@ private void testCreatePartitionedBucketedTableAsFewRows(Session session, HiveSt
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test
+    public void testCreateBucketedSortedTable()
+    {
+        String tableName = "test_create_bucketed_sorted_table";
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE " + tableName + " " +
+                "(id int, name varchar, ds varchar) " +
+                "WITH (" +
+                "bucketed_by = ARRAY[ 'id' ], " +
+                "bucket_count = 11, " +
+                "sorted_by = ARRAY[ '+name' ]" +
+                ")";
+        assertUpdate(createTable);
+    }
+
+    @Test(
+            expectedExceptions = RuntimeException.class,
+            expectedExceptionsMessageRegExp = "Sorted tables cannot be created using CREATE TABLE AS")
```

Comment:
Add a test for CREATE TABLE AS .. WITH NO DATA.
Should succeed.

Reply:
Per discussion with @electrum , this will require a larger change in the engine, which is unrelated to this PR. We could tackle that later.